apps/server/src/modules/user-login-migration/uc/user-login-migration.uc.ts

```diff
@@ -113,14 +120,72 @@ export class UserLoginMigrationUc {
 					targetSystemId
 				);
 
-				this.logger.debug(new SchoolMigrationSuccessfulLoggable(schoolToMigrate, userLoginMigration));
+				this.logger.info(new SchoolMigrationSuccessfulLoggable(schoolToMigrate, userLoginMigration));
 			}
 		}
 
 		await this.userMigrationService.migrateUser(currentUserId, data.externalUser.externalId, targetSystemId);
 
-		this.logger.debug(new UserMigrationSuccessfulLoggable(currentUserId, userLoginMigration));
+		this.logger.info(new UserMigrationSuccessfulLoggable(currentUserId, userLoginMigration));
 
 		await this.authenticationService.removeJwtFromWhitelist(userJwt);
 	}
+
+	async forceMigration(
+		userId: EntityId,
+		email: string,
+		externalUserId: string,
+		externalSchoolId: string
+	): Promise<void> {
+		const user: User = await this.authorizationService.getUserWithPermissions(userId);
+		this.authorizationService.checkAllPermissions(user, [Permission.USER_LOGIN_MIGRATION_FORCE]);
+
+		const schoolAdminUsers: UserDO[] = await this.userService.findByEmail(email);
+
+		if (schoolAdminUsers.length === 0) {
+			throw new NotFoundLoggableException('User', { email });
+		}
+		if (schoolAdminUsers.length > 1) {
+			throw new UserLoginMigrationMultipleEmailUsersLoggableException(email);
+		}
+
+		const schoolAdminUser: UserDO = schoolAdminUsers[0];
+		// TODO Use new domain object to always have an id
+		if (!schoolAdminUser.id) {
+			throw new NotFoundLoggableException('User', { email });
+		}
+
+		const isAdmin = !!schoolAdminUser.roles.find((value: RoleReference) => value.name === RoleName.ADMINISTRATOR);
+		if (!isAdmin) {
+			throw new UserLoginMigrationInvalidAdminLoggableException(schoolAdminUser.id);
+		}
+
+		const activeUserLoginMigration: UserLoginMigrationDO | null =
+			await this.userLoginMigrationService.findMigrationBySchool(schoolAdminUser.schoolId);
+		if (activeUserLoginMigration) {
+			throw new UserLoginMigrationSchoolAlreadyMigratedLoggableException(activeUserLoginMigration.schoolId);
+		}
+
+		const userLoginMigration: UserLoginMigrationDO = await this.userLoginMigrationService.startMigration(
+			schoolAdminUser.schoolId
+		);
+
+		const school: LegacySchoolDo = await this.schoolService.getSchoolById(schoolAdminUser.schoolId);
+
+		const hasSchoolMigrated: boolean = this.schoolMigrationService.hasSchoolMigrated(
```

Comment:
If we don't have a user login migration for this school, the school should not be migrated... is there a case i have missed or does it make sense to test ist twice just to be sure?

Reply:
better be safe than sorry. for the unlikely case that a school is already migrated and the migration entity was deleted.